test/test_signal_mod.py

```diff
@@ -0,0 +1,144 @@
+    assert signal.minvalue == test_minvalue
+    assert signal.maxvalue == test_maxvalue
+    assert signal.bitsize == test_bitsize
+    assert signal.signed == test_is_signed
+    assert signal.float == test_is_float
+
+
+def test_print_callback():
+    test_name = "name_test"
+    test_string = (
+        "#ifndef name_test_callback\n"
+        "#        define name_test_callback NULL\n"
+        "#endif"
+    )
+
+    signal = Signal()
+    signal.name = test_name
+    str_result = signal.print_callback()
+    assert str_result == test_string
+
+
+def test_print_enum():
+    test_name = "name_test"
+    test_string = "        name_test"
+
+    signal = Signal()
+    signal.name = test_name
+    str_result = signal.print_enum()
+    assert str_result == test_string
+
+
+def test_print_para_macro():
+    signal = Signal()
+    signal.name = "test_name"
+
+    for signal_type, init_value in {"U8": 255, "U16": 65000, "U32": 101, "F": 42.65}.items():
+        signal.type = signal_type
+        signal.init_value = init_value
+
+        last = False
+        test_string_not_last = f"        CAN_PARA_MACRO(test_name, sizeof({signal_type}), " \
+                               f"{init_value}, test_name_callback)"
+        str_result_not_last = signal.print_para_macro(last)
+
+        last = True
+        test_string_last = (
+            f"        CAN_PARA_MACRO(test_name, sizeof({signal_type}), "
+            f"{init_value}, test_name_callback),\n"
+            f"\n"
+            f"#ifdef E92_USE_CUSTOM_MEMSTRUCT\n"
+            f"/*******************************************************************/\n"
+            f"/*                           CUSTOM CAN                            */\n"
+            f"/*******************************************************************/\n"
+            f"        CUSTOM_CAN_SIG,\n"
+            f"#endif"
+        )
+        str_result_last = signal.print_para_macro(last)
+
+        assert str_result_not_last == test_string_not_last
+        assert str_result_last == test_string_last
+
+
+def test_print_definition():
+    signal = Signal()
+    signal.name = "test_name"
+
+    for last in [False, True]:
+        for little_endian in [True, False]:  # [LITTLE, BIG]
+            for byte_pos in [0, 8, 16, 24, 32, 40, 48, 56]:
+                comma_separator = '' if last else ','
+                byte_pos_str = f"{byte_pos}" if not little_endian else \
+                               f"({byte_pos})|(CANFRM_LITTLE_ENDIAN)"
+                test_string = (
+                    f"                        {{\n"
+                    f"                                test_name,       /* Signal ID */\n"
+                    f"                                {byte_pos_str}  /* Byte Position */\n"
+                    f"                        }}{comma_separator}\n"
+                )
+
+                str_result = signal.print_definition(byte_pos, last, little_endian)
+                assert str_result == test_string
```

Comment:
```suggestion
                assert str_result == test_string, f'Signal definition wrong: last = {last}, little_endian = {little_endian}, byte_pos = {byte_pos}'
```

Ajouter un message d'erreur custom pourrait aider étant donné que ce serait intéressant de savoir quelles valeurs d'entrée ont fait planter le test.